packages/orm/src/EntityManager.ts

```diff
@@ -618,39 +612,40 @@ export class EntityManager<C extends HasKnex = HasKnex> {
 
     this._isFlushing = true;
 
-    this.contexty = new Contexty();
-    this.contexty.create();
-
     const entitiesToFlush: Entity[] = [];
     let pendingEntities = this.entities.filter((e) => e.isPendingFlush);
 
-    // We need to split the thread so that Node's executionAsyncId changes and in turn
-    // Contexty give us a context that's distinct from whatever called flush
-    await new Promise((res) => setTimeout(res, 0));
-    const context = this.contexty.create();
-
     try {
       while (pendingEntities.length > 0) {
-        context.flushSecret = this.flushSecret;
-        const todos = sortEntities(pendingEntities);
-
-        // add objects to todos that have reactive hooks
-        await addReactiveAsyncDerivedValues(todos);
-        await addReactiveValidations(todos);
-
-        // run our hooks
-        await beforeDelete(this.ctx, todos);
-        // We defer doing this cascade logic until flush() so that delete() can remain synchronous.
-        await cascadeDeletesIntoRelations(todos);
-        await beforeFlush(this.ctx, todos);
-        recalcDerivedFields(todos);
-        await recalcAsyncDerivedFields(this, todos);
-        await validate(todos);
-        await afterValidation(this.ctx, todos);
-
-        entitiesToFlush.push(...pendingEntities);
-        pendingEntities = this.entities.filter((e) => e.isPendingFlush && !entitiesToFlush.includes(e));
-        this.flushSecret += 1;
+        await new Promise((resolve, reject) => {
+          currentFlushSecret.run({ flushSecret: this.flushSecret }, async () => {
```

Comment:
They don't technically support this `run` callback being async/a promise, so we use line 620 to promsify this callback.